dubbo-config/dubbo-config-spring/src/main/java/com/alibaba/dubbo/config/spring/AnnotationBean.java

```diff
@@ -288,6 +296,73 @@ private Object refer(Reference reference, Class<?> referenceClass) { //method.ge
         return referenceConfig.get();
     }
 
+    /**
+     * refer 方法的公开版，将注解参数Reference换为 ReferenceConf 类
+     * @param reference
+     * @param referenceClass
+     * @return
+     */
+    public Object referNew(RefConf reference, Class<?> referenceClass) { //method.getParameterTypes()[0]
+        String interfaceName;
+
+        if (! "".equals( reference.interfaceName)) {
+            interfaceName =  reference.interfaceName;
+        } else if (! void.class.equals( reference.interfaceClass)) {
+            interfaceName =  reference.interfaceClass.getName();
+        } else if (referenceClass.isInterface()) {
+            interfaceName = referenceClass.getName();
+        } else {
+            throw new IllegalStateException("The @Reference undefined interfaceClass or interfaceName, and the property type " + referenceClass.getName() + " is not a interface.");
+        }
+        String key =  reference.group + "/" + interfaceName + ":" +  reference.version;

```

Comment:
Please see com.alibaba.dubbo.common.URL#getServiceKey , you should check property value if exists.